repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/sql/QueryFolderPreProcessor.java

```diff
@@ -80,12 +91,211 @@ private void processCreate(AtlasStruct entity) throws AtlasBaseException {
         entity.setAttribute(QUALIFIED_NAME, createQualifiedName(collectionQualifiedName));
     }
 
-    private void processUpdate(AtlasStruct entity, AtlasVertex vertex) {
-        String vertexQnName = vertex.getProperty(QUALIFIED_NAME, String.class);
+    private void processUpdate(AtlasEntity entity, AtlasVertex vertex, EntityMutationContext context) throws AtlasBaseException {
+        AtlasEntityType entityType      = typeRegistry.getEntityTypeByName(entity.getTypeName());
+        AtlasObjectId newParentAttr     = (AtlasObjectId) entity.getRelationshipAttribute(PARENT_ATTRIBUTE_NAME);
+        String relationshipType         = AtlasEntityUtil.getRelationshipType(newParentAttr);
+        AtlasAttribute parentAttribute  = entityType.getRelationshipAttribute(PARENT_ATTRIBUTE_NAME, relationshipType);
+        AtlasObjectId currentParentAttr = (AtlasObjectId) entityRetriever.getEntityAttribute(vertex, parentAttribute);
+        //Qualified name of the folder will not be updated if parent attribute is not changed
+        String folderQualifiedName      = vertex.getProperty(QUALIFIED_NAME, String.class);
+        entity.setAttribute(QUALIFIED_NAME, folderQualifiedName);
+
+        //Check if parent attribute is changed
+        if (parentAttribute.getAttributeType().areEqualValues(currentParentAttr, newParentAttr, context.getGuidAssignments())) {
+            return;
+        }
+
+        AtlasVertex currentParentVertex         = entityRetriever.getEntityVertex(currentParentAttr);
+        AtlasVertex newParentVertex             = entityRetriever.getEntityVertex(newParentAttr);
+
+        if (currentParentVertex == null || newParentVertex == null) {
+            LOG.warn("Current or new parent vertex is null");
+            return;
+        }
+
+        String currentCollectionQualifiedName   = currentParentVertex.getProperty(getCollectionPropertyName(currentParentVertex), String.class);
+        String newCollectionQualifiedName       = newParentVertex.getProperty(getCollectionPropertyName(newParentVertex), String.class);
+        String updatedParentQualifiedName       = newParentVertex.getProperty(QUALIFIED_NAME, String.class);
+
+        if (StringUtils.isEmpty(newCollectionQualifiedName) || StringUtils.isEmpty(currentCollectionQualifiedName)) {
```

Comment:
Shouldn't we thrown an error here to let the client know something went wrong - otherwise client might assume query moved successfully

Reply:
Yes, addrressed